src/libmongoc/tests/mock_server/mock-server.c

```diff
@@ -981,49 +982,19 @@ _mock_server_receives_msg (mock_server_t *server, uint32_t flags, ...)
    return request;
 }
 
-
-/*--------------------------------------------------------------------------
- *
- * mock_server_receives_bulk_msg --
- *
- *       Pop a client OP_MSG request if one is enqueued, or wait up to
- *       request_timeout_ms for the client to send a request. Pass
- *       `msg_pattern`, which is matched to the series of exactly `n_doc`
- *       documents in the request, regardless of section boundaries.
- *
- * Returns:
- *       A request you must request_destroy, or NULL if the request does not
- *       match.
- *
- * Side effects:
- *       Logs and aborts if the current request is not an OP_MSG matching
- *       flags and expected pattern and number of documents.
- *
- *--------------------------------------------------------------------------
- */
 request_t *
-mock_server_receives_bulk_msg (mock_server_t *server,
-                               uint32_t flags,
-                               const bson_t *msg_pattern,
-                               const bson_t *doc_pattern,
-                               size_t n_docs)
+_mock_server_receives_single_msg (mock_server_t *server,
```

Comment:
This function is just a special case of `mock_server_receives_msg` and does not seem necessary. Please remove and replace calls to `_mock_server_receives_single_msg (server, flags, doc)` with `mock_server_receives_msg (server, flags, doc)`.

Reply:
Interestingly, this causes test failure in "/Topology/hello/versioned_api/single"; going through it again, I am not seeing the bug yet.

Reply:
I've added this to:
https://jira.mongodb.org/browse/CDRIVER-4280